parser_test.go

```diff
@@ -2241,19 +2242,24 @@ func (s *testParserSuite) TestView(c *C) {
 	c.Assert(ok, IsTrue)
 	c.Assert(v.Select.Text(), Equals, "select * from t")
 
-	src := `CREATE OR REPLACE ALGORITHM = UNDEFINED DEFINER = 'root@localhost'
+	src := `CREATE OR REPLACE ALGORITHM = UNDEFINED DEFINER = root@localhost
                   SQL SECURITY DEFINER
-			      VIEW V(a,b,c) AS select c,d,e from t
+			      VIEW V(a,b,c) AS select c,d,e from t 
                   WITH CASCADED CHECK OPTION;`
 
 	var st ast.StmtNode
-	st, err = p.ParseOneStmt(src,"","")
+	st, err = p.ParseOneStmt(src, "", "")
 	c.Assert(err, IsNil)
 	v, ok = st.(*ast.CreateViewStmt)
 	c.Assert(ok, IsTrue)
 	c.Assert(v.OrReplace, IsTrue)
 	c.Assert(v.Algorithm, Equals, ast.AlgorithmUndefined)
-	c.Assert(v.Definer, NotNil)
+	c.Assert(v.Definer.Username, Equals, "root")
+	c.Assert(v.Definer.Hostname, Equals, "localhost")
+	c.Assert(v.Cols[0], Equals, model.CIStr{"a", "a"})
+	c.Assert(v.Cols[1], Equals, model.CIStr{"b", "b"})
+	c.Assert(v.Cols[2], Equals, model.CIStr{"c", "c"})
+	c.Assert(strings.TrimSpace(v.Select.Text()), Equals, "select c,d,e from t")
```

Comment:
Why do we TrimSpace here?
If there are redundant spaces, we need to change parser.y to extract the exact sql text.

Reply:
Just done yet...